app/setup.py

```diff
@@ -109,7 +98,9 @@ def __getattr__(self, attr: str) -> Any:
     def write(self, content: str) -> None:
         for key in _SENSITIVE_KEYS:
             if secret := getattr(config, key):
```

Comment:
I don't really have an opinion on whether it is desirable, but I wonder if this should be something like:

```suggestion
for key, value in config:
    if isinstance(key, SecretStr):
```

so as to avoid duplicating the secret strings separately.